frontend/dockerfile/dockerfile_test.go

```diff
@@ -4685,6 +4688,100 @@ COPY foo foo2
 	require.Equal(t, expected, actual)
 }
 
+func testFrontendSubrequests(t *testing.T, sb integration.Sandbox) {
+	f := getFrontend(t, sb)
+
+	c, err := client.New(context.TODO(), sb.Address())
+	require.NoError(t, err)
+	defer c.Close()
+
+	dockerfile := []byte(`
+FROM scratch
+COPY Dockerfile Dockerfile
+`)
+
+	if gf, ok := f.(*gatewayFrontend); ok {
+		dockerfile = []byte(fmt.Sprintf("#syntax=%s\n\n%s", gf.gw, dockerfile))
+	}
+
+	dir, err := tmpdir(
+		fstest.CreateFile("Dockerfile", dockerfile, 0600),
+	)
+	require.NoError(t, err)
+	defer os.RemoveAll(dir)
+
+	called := false
+
+	frontend := func(ctx context.Context, c gateway.Client) (*gateway.Result, error) {
+		res, err := c.Solve(ctx, gateway.SolveRequest{
+			FrontendOpt: map[string]string{
+				"requestid":     "frontend.subrequests.describe",
+				"frontend.caps": "moby.buildkit.frontend.subrequests",
```

Comment:
I think it'll be nice to have a client helper to make this request, like `c.Describe(ctx, "dockerfile.v0")`

Reply:
I could make a package that has functions that take `gateway.Client` as argument. As anyone can define requests for their frontend don't want to hardcode any of it to the main client package.

Reply:
Yeah that makes sense, I just thought `frontend.subrequests.describe` is a subrequest most frontends will want to implement.

Reply:
From frontend side this is already reusable and in a separate package. But I can make the caller side more reusable as well.